sys/include/fs.h

```diff
@@ -0,0 +1,405 @@
+     * @param[in] rdev  Device number (if the call creates a device)
+     *
+     * @return 0 on success,
+     * @return TODO error codes
+     */
+    int (*mknod)(const char *path, mode_t mode, dev_t rdev);
+
+    /**
+     * @brief   Opens a file.
+     *
+     * Flags can be passed via *info*::flags
+     *
+     * @param[in] path      Path to the file.
+     * @param[in,out] info  File information handed by filesystem API
+     *
+     * @return positive file id on success,
+     * @return TODO error codes
+     */
+    fid_t (*open)(const char *path, fs_file_info_t *info);
+
+    /**
+     * @brief   Opens a directory.
+     *
+     * @param[in] path      Path to the directory.
+     * @param[in,out] info  File information handed by the filesystem API.
+     *
+     * @return positive file id on success,
+     * @return TODO error codes
+     */
+    fid_t (*opendir)(const char *path, fs_file_info_t *info);
+
+    /**
+     * @brief   Read from a file
+     *
+     * @param[in] path      Path to file.
+     * @param[out] buf      Buffer to be filled.
+     * @param[in] size      Size of *buf*.
+     * @param[in] offset    Offset in *buf*.
+     * @param[in,out] info  File information handed by the filesystem API.
+     */
+    int (*read)(const char *path, char *buf, size_t size, off_t offset,
+                fs_file_info_t *info);
+
+    int (*readdir)(const char *path, void *); // TODO
+
+    /**
+     * TODO
+    int (*readlink)(const char *, char *, size_t);
+    int (*release)(const char *);
+    int (*releasedir)(const char *);
+    int (*rename)(const char *, const char *);
+    int (*rmdir)(const char *);
+    int (*stat)(const char*, struct stat *);
+    int (*statfs)(const char *, struct statvfs *);
+    int (*symlink)(const char *, const char *);
+    int (*truncate)(const char *, off_t);
+    int (*unlink)(const char *);
+    int (*utimens)(const char *, const struct timespec tv[2]);
+    int (*write)(const char *, const char *, size_t, off_t);
+    int (*write_buf)(const char *, off_t off);
+     */
+} fs_op_table_t;
+
+
+/**
+ * @brief      Mounts the given |device| at |mount_point|.
+ * @param[in]  mount_point Path where the device should be mounted
+ * @param[in]  device      Device to be mounted
+ *                         TODO: What kind of value is this
+ * @param[in]  type        The desired file system
+ * @return 0, on success
+ * @return -ENAMETOOLONG, if the given mount point exceeds the maximum length
+ * @return -ENOMEM, if more than FS_TABLE_SIZE filesystems are mounted
+ */
+int fs_mount(const char *mount_point, int device, fs_type_t type);
+
+/**
+ * @brief   Unmounts the filesystem at the given path OR unmounts the given device
+ * @param  mount_point Path to be unmounted
+ * @param  device      Device to be unmounted
+ *                     TODO: What kind of value is this
+ * @return 0, on success
+ * @return ENOENT, TODO
+ * @return EINVAL, TODO
+ */
+int fs_unmount(const char *mount_point, int device);
```

Comment:
Does `fs_unmount` actually need both `mount_point` and `device`? Wouldn't just the mount_point suffice, as the device is contained in `fs_table_t`?